cucumber-messages/dotnet/messages.proto

```diff
@@ -396,6 +399,26 @@ message PickleStepArgument {
   }
 }
 
+////// TestCases
+
+message TestCase {
+  string id = 1;
+  string pickleId = 2;
+  repeated TestStep steps = 3;
+  
+  message TestStep {
+    string id = 1;
+    // Pointer to the PickleStep (if derived from a PickleStep)
+    string pickleStepId = 2;
+    // Pointer to all the matching StepDefinitions (if derived from a PickleStep)
+    repeated string stepDefinitionId = 3;
```

Comment:
Also if we are going to use ids here rather than the previous actionLocation / sourceLocation do we also need additional messages to expose the step / hook definitions to formatters? Currently those are only part of the Command messages